scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/synchronization/AOSynchronization.java

```diff
@@ -0,0 +1,824 @@
+import org.objectweb.proactive.Body;
+import org.objectweb.proactive.EndActive;
+import org.objectweb.proactive.InitActive;
+import org.objectweb.proactive.RunActive;
+import org.objectweb.proactive.Service;
+import org.objectweb.proactive.core.body.request.Request;
+import org.objectweb.proactive.core.mop.MethodCallExecutionFailedException;
+import org.objectweb.proactive.extensions.annotation.ActiveObject;
+import org.objectweb.proactive.utils.StackTraceUtil;
+import org.ow2.proactive.scheduler.common.task.TaskId;
+import org.ow2.proactive.scheduler.util.TaskLogger;
+
+import groovy.lang.GroovyShell;
+import jdbm.PrimaryHashMap;
+import jdbm.RecordManager;
+import jdbm.RecordManagerFactory;
+
+
+/**
+ * @author ActiveEon Team
+ * @since 26/03/2018
+ */
+@ActiveObject
+public class AOSynchronization implements RunActive, InitActive, EndActive, SynchronizationInternal {
+
+    private static final Logger logger = Logger.getLogger(AOSynchronization.class);
+
+    private File statusFile;
+
+    private File statusFileDirectory;
+
+    private static final String DEFAULT_STORE_SESSION_NAME = "SchedulerStore";
+
+    public static final String STATUS_RECORD_NAME = "STORE";
+
+    private HashMap<String, Channel> inMemoryChannels;
+
+    private PrimaryHashMap<String, Channel> persistedChannels;
+
+    private Queue<Request> waitUntilQueue = new ArrayDeque<>();
+
+    private CompilerConfiguration compilerConfiguration;
+
+    private GroovyShell shell;
+
+    protected RecordManager recMan;
+
+    public AOSynchronization() {
+
+    }
+
+    public AOSynchronization(String statusFileDirectoryPath) {
+        compilerConfiguration = new CompilerConfiguration();
+        compilerConfiguration.setTargetBytecode(CompilerConfiguration.JDK8);
+        shell = new GroovyShell(this.getClass().getClassLoader(), compilerConfiguration);
+        this.statusFileDirectory = new File(statusFileDirectoryPath);
+        if (!statusFileDirectory.exists()) {
+            statusFileDirectory.mkdirs();
+        }
+        statusFile = new File(statusFileDirectory, DEFAULT_STORE_SESSION_NAME);
+        inMemoryChannels = new HashMap<>();
+
+    }
+
+    private boolean init(boolean firstAttempt) {
+        close();
+        try {
+            logger.info("Loading Record Manager from file : " + statusFile);
+            recMan = RecordManagerFactory.createRecordManager(statusFile.getCanonicalPath());
+            persistedChannels = recMan.hashMap(STATUS_RECORD_NAME);
+            logger.info("Content of persisted store : " + persistedChannels);
+            // This empty loop triggers InvalidClassException in case of serial version uid problems
+            for (Map.Entry<String, Channel> tempChannels : persistedChannels.entrySet())
+                ;
+            recMan.commit();
+            if (!firstAttempt) {
+                logger.info("Loading of job database successful after clean.");
+            }
+        } catch (Throwable e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1181)